model_analyzer/entrypoint.py

```diff
@@ -33,6 +33,8 @@
 from .record.gpu_free_memory import GPUFreeMemory
 from .record.gpu_used_memory import GPUUsedMemory
 from .record.gpu_utilization import GPUUtilization
+from .record.cpu_used_memory import CPUUsedMemory
```

Comment:
All of these imports are creating huge dependency chains. I shouldn't need to edit this top level file if I create a new record type. All record types should use the same interface and this top level file should reference only the single interface. Not the individual instances of the records should be abstracted away.

Reply:
I think this will likely move into the input parsing side of the code, because we want the user to control what metrics they measure. However, I don't see how we can implement this without some dispatching code where all of these imports are in the same place.

Reply:
We can probably import all the records. For example, we can use `from .record import *` so that we do not need to update this list every time. But I guess still the dependencies will exist.

Reply:
I like the idea, though I think doing `import *` is recommended against in general right?

Reply:
Additionally, this moves the problem to the line 
```
monitoring_metrics = [
        PerfThroughput, GPUUtilization, GPUUsedMemory, GPUFreeMemory,
        CPUUsedMemory, CPUFreeMemory
    ]
```

Reply:
I see...

Reply:
It is not the responsibility of the top level application to know what the monitoring metrics are. What you needs is a Object Constructor design pattern. A class that constructs each type and breaks the dependency on the individual type.

Reply:
A single class per type or a factory?